src/main/java/io/cdap/plugin/aws/sqs/source/SqsSourceConfig.java

```diff
@@ -0,0 +1,440 @@
+    }
+  }
+
+  private void validateRegion(FailureCollector collector) {
+    if (containsMacro(PROPERTY_REGION)) {
+      return;
+    }
+
+    getRegion(collector);
+  }
+
+  private void validateInterval(FailureCollector collector) {
+    if (containsMacro(PROPERTY_INTERVAL)) {
+      return;
+    }
+
+    if (interval < 0) {
+      collector.addFailure(String.format("Invalid interval '%d'.", interval),
+        "Ensure the interval is not less than 0")
+        .withConfigProperty(PROPERTY_INTERVAL);
+    }
+  }
+
+  private void validateWaitTime(FailureCollector collector) {
+    if (containsMacro(PROPERTY_WAIT_TIME)) {
+      return;
+    }
+
+    if (waitTime < 1 || waitTime > MAX_WAIT_TIME) {
+      collector.addFailure(String.format("Invalid wait time '%d'.", waitTime),
+        String.format("Ensure the wait time is 1 or at most '%d'", MAX_WAIT_TIME))
+        .withConfigProperty(PROPERTY_WAIT_TIME);
+    }
+  }
+
+  private void validateMaxNumberOfMessages(FailureCollector collector) {
+    if (containsMacro(PROPERTY_NUMBER_OF_MESSAGES)) {
+      return;
+    }
+
+    if (numberOfMessages < 1 || numberOfMessages > MAX_MESSAGES_LIMIT) {
+      collector.addFailure(String.format("Invalid number of messages '%d'.", numberOfMessages),
+        String.format("Ensure the number of messages is 1 or at most '%d'", MAX_MESSAGES_LIMIT))
+        .withConfigProperty(PROPERTY_NUMBER_OF_MESSAGES);
+    }
+  }
+
+  private void validateSchema(FailureCollector collector) {
+    if (containsMacro(PROPERTY_SCHEMA)) {
+      return;
+    }
+
+    if (Utility.isNullOrEmpty(schema)) {
+      collector.addFailure("Schema must be specified.", null)
+        .withConfigProperty(PROPERTY_SCHEMA);
+    }
+  }
+
+  private void validateFormat(FailureCollector collector) {
+    if (containsMacro(PROPERTY_SCHEMA) || containsMacro(PROPERTY_FORMAT)) {
+      return;
+    }
+
+    if (Utility.isNullOrEmpty(schema)) {
+      return;
+    }
+
+    Schema messageSchema = null;
+    try {
+      messageSchema = parseSchema();
+    } catch (Exception e) {
+      collector.addFailure(e.getMessage(), null)
+        .withConfigProperty(PROPERTY_SCHEMA);
+      collector.getOrThrowException();
+    }
+
+    // if format is empty, there must be just a single message field of type bytes or nullable
+    // types.
+    if (Utility.isNullOrEmpty(format)) {
+      List<Schema.Field> messageFields = messageSchema.getFields();
+      if (messageFields.size() > 1) {
+        List<String> fieldNames = messageFields.stream().map(o -> o.getName()).collect(Collectors.toList());
+        collector.addFailure(String.format("Without a format, the schema must contain just a single message field " +
+            "of type bytes or nullable bytes. Found %s message fields (%s).", messageFields.size(),
+          Joiner.on(',').join(fieldNames)), null)
+          .withConfigProperty(PROPERTY_SCHEMA);
+        return;
+      }
+      Schema.Field messageField = messageFields.get(0);
```

Comment:
What if the schema has 0 fields? We should check that before getting the first element

Reply:
Done

Reply:
If messageFields is empty then it will never reach this line of code. This is because in case of messageFields is empty then following code will take care of that,

```
    Schema messageSchema = validateAndGetSchema(collector);
    if (messageSchema == null) {
      return;
    }
```

Reply:
I looks like the `validateAndGetSchema` function will still return a non-null value if there are no fields (see below). Is that intentional?

```
    Schema messageSchema = null;
    try {
      messageSchema = parseSchema();

      if (messageSchema.getFields().isEmpty()) {
        collector.addFailure("Schema must be specified.", null)
          .withConfigProperty(SqsSourceConstants.PROPERTY_SCHEMA);
      }
    } catch (Exception e) {...}
```